### PR TITLE
gpu: better logging for debugging

### PR DESCRIPTION
**Pull request by sharanyad**
### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [ ] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [ ] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/gpu/nvidia_gpu_manager_unix.go [hunk 1]
```diff
@@ -78,6 +79,7 @@ func (n *NvidiaGPUManager) Initialize() error {
 		n.SetGPUIDs(gpuIDs)
 		n.SetDevices()
 	}
+	seelog.Info("Config for GPU support is enabled, but GPU information is not found; continuing without it")
```
Comment (yunhee-l) [inline agent/gpu/nvidia_gpu_manager_unix.go]: wondering if this should be handled as a failure, for example when customer is expecting to use GPU, shouldn't the customer be notified of the misconfiguration in a more explicit way?

Comment (sharanyad) [reply agent/gpu/nvidia_gpu_manager_unix.go]: This config is not recommended for customers to add. This is a GPU instance level config, but we also let customers override it to true/false. If for some reason, a customer sets it to true on non GPU instance, there is value is not stopping agent and continuing to register as a normal instance. I can change the log level to error.

**Comment by sharanyad** [comment]
flakey windows functional test failure: 
`--- FAIL: TestOOMContainer (20.53s)`

`TestTaskLevelVolume` timed out for windows integ test, which seems unrelated. Will add a note and it could be investigated outside of this PR.

**Comment by sharanyad** [comment]
merging PR since failures are not related to the changes

**Pull request merged**